tests/python_functional/src/syslog_ng_ctl/driver_stats_handler.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+#############################################################################
+# Copyright (c) 2015-2018 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+import src.testcase_parameters.testcase_parameters as tc_parameters
+from src.syslog_ng_ctl.syslog_ng_ctl import SyslogNgCtl
+
+
+class DriverStatsHandler(object):
+    def __init__(self, group_type, driver_name):
+        if group_type == "destination":
+            statement_short_name = "dst"
+        elif group_type == "source":
+            statement_short_name = "src"
+
+        self.component = "{}.{}".format(statement_short_name, driver_name)
+        self.syslog_ng_ctl = SyslogNgCtl(tc_parameters.INSTANCE_PATH)
+
+    def build_query_pattern(self):
+        return self.build_pattern(delimiter=".")
+
+    def build_stats_pattern(self):
+        return self.build_pattern(delimiter=";")
+
+    def build_pattern(self, delimiter):
+        return "{}{}".format(
+            self.component,
+            delimiter,
+        )
+
+    def parse_result(self, result, result_type):
+        statistical_elements = ["memory_usage", "written", "processed", "dropped", "queued", "stamp"]
```

Comment:
Optional: It would be better to parse every line on its own, without having the assumption, that these elements will be the only valid cases. That would automatically follow the changes in the `stats` output.

Reply:
I'm not sure I understand what you want. We have to select those lines which contains the statistical elements. If I just try to convert all incoming lines it may parse wrong elements as well.
If I misunderstand your comment, it would be good to have a quick discussion about it.